CraftsmenLtd/BloodConnect
# Added Throttling → DynamoDB → DynamoDB Streams → EventBridge Pipe → SQS Workflow Integration for Blood Post Request System

Fixes #83


## What kind of change does this PR introduce (check at least one)


## How would you describe the changes in a changelog (1-2 lines)?

Implemented API throttling (10 requests/day/user), DynamoDB Stream integration with EventBridge pipe, and SQS queue setup for donor search workflow.

## Related Issues/PRs

- Built upon PR #177
- Implements the remaining workflow requirements from ticket #83

## How to test


Testing steps:
1. Create more than 10 blood donation requests with the same seekerId within 24 hours
   - Verify throttling error after 10th request
2. Create a new blood donation request
   - Verify entry in DynamoDB
   - Verify event capture in DynamoDB Stream
   - Verify message delivery to donor-search-queue via EventBridge pipe
3. Update an existing blood donation request
   - Verify update in DynamoDB
   - Verify event capture in Stream
   - Verify message in donor-search-queue

## Checklist

- [x] I have checked for potential conflicts with other branches or pull requests.

## Additional Notes

Infrastructure changes:
1. Added DynamoDB Stream configuration
2. Set up EventBridge pipe for event routing
3. Created SQS queues (main, retry, and DLQ)
4. Implemented necessary IAM roles and policies

Code changes:
1. Added request throttling in BloodDonationService
2. Enhanced error handling for throttling scenarios
3. Updated DynamoDB table configuration
4. Added EventBridge pipe configuration with filtering

## Deploy risk & follow up
